### ISSUE TITLE
Listing more than one body parameter.

### ISSUE THREAD

**Issue by Patskimoto** [Mermade/widdershins#29]
Apologies if this is not a widdershins issue but I am not 100% sure. I am trying to parse my swagger .json file to slate mark up and I can never get more than one body parameter to show in the parameters list for an endpoint. I understand the a request can only have one body but I want to list multiple params and simply have "In Body" next to it. However if I have more than one body param in my swagger.json and I run widdershins to convert it crashes.

I can make a single body param and reference it to an object but then the params show under "Body example" on the right hand slide of the slate doc and it only shows "body" under params on the left which I feel is a bit confusing. I could obviously just hard code the parameter details in but really trying to avoid that option.

This is my desired output:

![want](https://user-images.githubusercontent.com/14765526/28656497-b21844c8-72e5-11e7-816c-172737d43584.png)

But this is all I can do (one body param or widdershins fails to parse):

![get](https://user-images.githubusercontent.com/14765526/28656526-cda5231e-72e5-11e7-850e-569866f9d8a2.png)

Feel free to close if this is something outside of Widdershins control.

**Comment by MikeRalphson** [comment]
Unfortunately the limitation on only having one `in:body` parameter is part of the OpenAPI 2.0 specification. Though, that said, Widdershins should not crash in this case, so that's bad!

Do you have a sample json file I can test with (as a gist etc), otherwise I can mock one up if it's commercially sensitive?

An option to expand body parameter definitions is certainly something we can look at, I just need to think how it would work if the object had more levels of nesting...

**Comment by MikeRalphson** [comment]
Related to #26

**Comment by MikeRalphson** [comment]
@Patskimoto would you mind testing the latest version, to see if it improves things for you?

* Fixed the crash on multiple body parameters - still aborts, but cleanly now
* Objects in the body parameter / requestBody are now expanded by default
* A new 'schemas' section has been added with expanded properties of each top level schema
* Hyperlinks beween object use in parameters and responses and schema definitions

Let me know if you're using the npm package, and I will do a release.

**Comment by MikeRalphson** [comment]
Should be fixed in v2.1.1 - please let me know if not.

**Comment by fgsalomon** [comment]
Hi,

I'm not sure if it's the same issue but when using multiple request bodies with OpenAPI 3 widdershins it's not generating all the entries on the Parameters table.

For example when using this specification:
```yaml
openapi: "3.0.0"                          
info:                                     
  version: 1.0.0                          
  title: Swagger Hello                    
  license:                                
    name: MIT                             
servers:                                  
  - url: http://hello.swagger.io/v1                                                                 
paths:                                                                                              
  /hello:                                                                                           
    post:                                                                                            
      summary: hello world                                                                          
      operationId: hello                                                                            
      requestBody:                                                                                  
        content:                                                                                    
          application/json:                                                                         
              schema:                                                                               
                $ref: "#/components/schemas/Greetings"                                              
          multipart/form-data:                                                                      
               schema:                                                                              
                $ref: "#/components/schemas/GreetingsFormData"                                      
      responses:                                                                                    
        '200':
          description: "Greetings response"
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/GreetingsResponse" 
        default:                                                                                    
          description: unexpected error                                                             
          content:                                                                                  
            application/json:                                                                       
              schema:                                                                               
                $ref: "#/components/schemas/Error"                                                  
components:                                                                                         
  schemas:                                                                                          
    Greetings:                                                                                      
      required:                                                                                     
        - name                                                                                      
      properties:                                                                                   
        name:                                                                                       
          type: string                                                                              
        tag:                                                                                        
          type: string     
    GreetingsResponse:                                                                                      
      required:                                                                                     
        - name                                                                                      
      properties:                                                                                   
        name:                                                                                       
          type: string                                                                              
        tag:                                                                                        
          type: string  
    GreetingsFormData:                                                                              
      type: object                                                                                  
      properties:                                                                                   
        name:                                                                                       
          type: string                                                                              
    Error:                                                                                          
      required:                                                                                     
        - code                                                                                      
        - message                                                                                   
      properties:                                                                                   
        code:                                                                                       
          type: integer                                                                             
          format: int32                                                                             
        message:                                                                                    
          type: string                                                                              
```
I'd like to see both `Greetings` and `GreetingsFormData` on the Parameters table and only the first one is shown when generating the HTML with `api2html` (version 0.2.1).
